catalog/recursion-to-filter.md

```diff
@@ -0,0 +1,87 @@
+# Recursion to Filter
+
+## Relevant To
+
+
+* Replace Loop with Collection Closure Method
+* Replace Temp with Chain
+
+## Relates To
+
+* Recursion to Map
+* Recursion to Reduce
+* Tail Call Optimise Recursion
+
+## Languages
+
+All
+
+## Motivation
+
+Selectively filter a list of items into a new, smaller list of items, based
```

Comment:
The motivation need be what moving from recursion to a filter is better. The points I can think of are:

* It describes the action rather than the implementation of the action
* It separates the predicate logic from the filtering logic (providing greater opportunities for composability)
* It is not restricted by stack size in languages where tail call recursion is not supported